.github/workflows/build.yml

```diff
@@ -373,7 +373,7 @@ jobs:
 
       - name: Archive failed tests screenshots
         if: failure()
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v4
```

Comment:
Thanks to @filipefurtad0's PR https://github.com/openfoodfoundation/openfoodnetwork/pull/12702, it's observed that we need to remove the following lines because we only need to upload the failed test screenshots generated by system specs and don't need anything to be uploaded from engine specs. https://github.com/openfoodfoundation/openfoodnetwork/blob/da246380794894e704c60b59fc3010c690135dfd/.github/workflows/build.yml#L374-L381